tools/gulp/tasks/screenshots.ts

```diff
@@ -0,0 +1,190 @@
+import {task} from 'gulp';
+import {readdirSync, statSync, existsSync, mkdirSync} from 'fs';
+import {openScreenshotsCloudStorage, openFirebaseScreenshotsDatabase} from '../task_helpers';
+import * as path from 'path';
+import * as admin from 'firebase-admin';
+const request = require('request');
+const imageDiff = require('image-diff');
+
+const SCREENSHOT_DIR = './screenshots';
+const FIREBASE_FILELIST = 'screenshot/filenames';
+const FIREBASE_REPORT = 'screenshot/reports';
+
+/** Task which upload screenshots generated from e2e test. */
+task('screenshots', () => {
+  let prNumber = process.env['TRAVIS_PULL_REQUEST'];
+  if (prNumber) {
+    let database = openFirebaseScreenshotsDatabase();
+    return getScreenFilenames(database)
+      .then((filenames: string[]) => downloadAllGolds(filenames, database, prNumber))
+      .then((results: boolean) => updateResult(database, prNumber, results))
+      .then((result: boolean) => updateGithubStatus(result, prNumber))
+      .then(() => setScreenFilenames(database, prNumber))
+      .then(() => uploadScreenshots(prNumber, 'diff'))
+      .then(() => uploadScreenshots(prNumber, 'test'))
+      .then(() => updateTravis(database, prNumber))
+      .then(() => database.goOffline(), () => database.goOffline());
+  }
+});
+
+function updateFileResult(database: admin.database.Database, prNumber: string,
+                          filenameKey: string, result: boolean): admin.Promise<void>{
+  return database.ref(FIREBASE_REPORT).child(`${prNumber}/results/${filenameKey}`).set(result);
+}
+
+function updateResult(database: admin.database.Database, prNumber: string,
+                      result: boolean): admin.Promise<void> {
```

Comment:
This seems to return a `boolean` instead of `void`. Can't you remove those `admin.Promise` return types and just let TypeScript detect it properly? (same for others)